codegen/gateway.go

```diff
@@ -184,6 +192,7 @@ func NewTChannelClientSpec(
 	cspec.ExposedMethods = make(map[string]string, len(exposedMethods))
 	reversed := make(map[string]string, len(exposedMethods))
 	for key, val := range exposedMethods {
+		key = correctMethodNaming(key)
```

Comment:
We should not do this, this creates confusion and inconsistency between config and generated code, i.e., I configured my method name to be `TestUrl`(which is wrong) but the codegen generates a method `TestURL`. The user should provide `TestURL` in the configuration at first place.